toptal/picasso
# fix(Popper): fixed Popper overflow inside Modal

[SPT-2231]

### Description

When some component that uses `Popper` under the hood is placed inside the `Modal` and `Modal` is scrolled, then `Popper` overflows `Modal` component.
To fix this we can set `boundariesElement: 'scrollParent'` for `Popper`, but it's dangerous to set this globally, so right now it's safer to set this modifier only when `Popper` is inside `Modal`.
To achieve this i added `ModalContext` that is used inside `Popper` to determine if it's open inside `Modal`

### How to test

- Open https://picasso.toptal.net/SPT-2231-fix-popper-overflow-inside-modals/?path=/story/overlays-modal--modal&cb95db0c05bddea59535cd3e74fbeb0b3e4ce346/?path=/story/overlays-modal--modal , first example
- Add various items to the modal content to display modal scroll bar
- Add about 15 items to the Form.Select component
- Open this select and scroll modal content
- Check that popup of the Select will be hidden when Select is overflowed
- Check the same behavior for DatePicker in the same Modal

### Videos

Before:
https://user-images.githubusercontent.com/9210534/146951922-cbe8d8e2-43e5-45fc-8580-3a901b699713.mov

After:
https://user-images.githubusercontent.com/9210534/146951685-c0753925-1ac7-4965-bcba-192b610002eb.mov


### Review

- [n/a] Annotate all `props` in component with documentation
- [n/a] Create `examples` for component

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and
1. You have no extra requests.
2. You have optional requests.
   1. Add `nit: ` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because
1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:
1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[SPT-2231]: https://toptal-core.atlassian.net/browse/SPT-2231?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
